### ISSUE TITLE
Ignore non API controllers for Swagger document generation

### ISSUE THREAD

**Issue by brgrz** [domaindrivendev/Swashbuckle.AspNetCore#2031]
Swashbuckle requires plastering controllers/actions with 

`[ApiExplorerSettings(IgnoreApi = true)]`

to ignore including them in document generation.

However, API controllers already typically inherit `ControllerBase` class while controllers with views inherti the `Controller` class.
Also, API controllers are most likely already decorated with the `ApiController` attribute.

Imo this should be enough for Swashbuckle to only pick the ones the either inherit ControllerBase and/or those the have the `[ApiController]` decorations. 

Decorating those that we don't want included with `[ApiExplorerSettings(IgnoreApi = true)]` seems redundant.

**Comment by domaindrivendev** [comment]
What you’re looking for is including/omitting actions by convention - the convention in your case being based on the controller inheritance hierarchy. Swashbuckle is built on top of ApiExplorer, which already has a mechanism for this, and so you just need to wire up a simple ApiExplorer convention as described in the following link:

https://github.com/domaindrivendev/Swashbuckle.AspNetCore#omit-actions-by-convention

**Comment by domaindrivendev** [comment]
For example:

```csharp
public class ApiExplorerConvention : IActionModelConvention
{
    public void Apply(ActionModel action)
    {
        action.ApiExplorer.IsVisible = action.Controller.ControllerType.BaseType == typeof(ControllerBase);
    }
}
```csharp

**Comment by brgrz** [comment]
I ran into this in an MVC project where view controller actions were missing verbs and ran into 

`SwaggerGeneratorException: Ambiguous HTTP method for action - StaticPageController.Default. Actions require an explicit HttpMethod binding for Swagger/OpenAPI 3.0`

So I decorated them with verbs and then those actions made up into Swagger doc. Which by default I don't think makes sense, or does it?

I wish the defaults were such that view controllers are excluded. Without having to decorate anything.

I'm not pushing this, it just makes more sense to me.
